action/protocol/staking/contractstaking/indexer.go

```diff
@@ -91,3 +96,105 @@ func (s *Indexer) BucketTypes() ([]*BucketType, error) {
 	}
 	return bts, nil
 }
+
+// PutBlock puts a block into indexer
+func (s *Indexer) PutBlock(ctx context.Context, blk *block.Block) error {
+	// new dirty cache for this block
+	// it's not necessary to use thread safe cache here, because only one thread will call this function
+	// and no update to cache will happen before dirty merge to clean
+	dirty := newContractStakingDirty(s.cache.Unsafe())
```

Comment:
it doesn't hurt much to use CacheSafe, thus, Cache and CacheSafe could be merged into one.

Reply:
ok, removed CacheSafe and move the mutex into Cache